addon/lib/metrics.js

```diff
@@ -75,8 +75,8 @@ module.exports = {
 
     this.maybePingTelemetry();
 
-    Events.on(EVENT_SEND_METRIC, this.onExperimentPing);
-    Events.on(EVENT_RECEIVE_VARIANT_DEFS, this.onReceiveVariantDefs);
+    Events.on(EVENT_SEND_METRIC, this.onExperimentPing.bind(this));
+    Events.on(EVENT_RECEIVE_VARIANT_DEFS, this.onReceiveVariantDefs.bind(this));
```

Comment:
This won't work because we use `Events.off` to disconnect the functions further down